hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/client/DefaultCertificateClient.java

```diff
@@ -162,14 +154,6 @@ public abstract class DefaultCertificateClient implements CertificateClient {
     loadAllCertificates();
   }
 
-  public synchronized void setCertificateId(String certId) {
```

Comment:
@fapifta , thanks for this cleanup work. 

I would suggest keep this setCertificateId API.  When default certID is changed, loadAllCertificates() need be called to update certificateMap and ca, root ca cert ID too.  This whole process is synchronized to avoid any mix use of old and new data.  

With the remove of  setCertificateId, it's error prone.  In the commit history, loadAllCertificates call is missed in reloadKeyAndCertificate and synchronized key word is missed on signAndStoreCertificate.  And synchronized on signAndStoreCertificate is not appropriate, for signAndStoreCertificate includes the call to SCM to get certificates, any network vibration will block the CertificateClient. 

So my overall suggestion is bring back the setCertificateId function. It may be not necessary be part of CertificateClient interface, we can keep it in DefaultCertificateClient.  

Besides setCertificateId function, other parts look good to me.

Reply:
Thank you @ChenSammi for the suggestion, after thinking it through, it is really a better approach to kepp the method, so that the changes are more safer. I re-added the method as a private method, and updated the code to use that whenever the certSerialID is updated.